src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -1211,6 +1211,56 @@ class FunctionBinaryArithmetic : public IFunction
         return ColumnArray::create(res, typeid_cast<const ColumnArray *>(arguments[0].column.get())->getOffsetsPtr());
     }
 
+    ColumnPtr executeArrayWithNumericImpl(const ColumnsWithTypeAndName & args, const DataTypePtr & result_type, size_t input_rows_count) const
+    {
+        ColumnsWithTypeAndName arguments = args;
+        if (isNumber(args[0].type)) /// We should have scalar as a second argument
+            std::swap(arguments[0], arguments[1]);
```

Comment:
But can we do this for division? Multiplication is ok, it is commutative. But division is not :)